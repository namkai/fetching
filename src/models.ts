interface Height {
  imperial: string;
  metric: string;
}

interface Weight {
  imperial: string;
  metric: string;
}

export interface Breed {
  id: number;
  bred_for: string;
  breed_group: string;
  country_code: string;
  height: Height;
  life_span: string;
  name: string;
  origin: string;
  temperament: string;
  weight: Weight;
}
