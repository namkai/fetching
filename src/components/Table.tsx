import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
`

interface TableProps {

}

const enum

export default function Table(): React.FC<TableProps> {
  const []
  return (
    <Container>
      Table
    </Container>
  )
}