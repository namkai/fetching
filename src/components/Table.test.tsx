import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Breed } from "../models";
import Table from "./Table";

const data: Partial<Breed>[] = [
  { id: 231, name: "Dog", life_span: "10 - 12" },
  { id: 123, name: "Other Dog", life_span: "6 - 30" },
];
const keys: Array<keyof Breed> = ["id", "name", "life_span"];

describe("Table", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Table keys={keys} data={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should return a header with the specified columns", () => {
    const { getByText } = render(<Table keys={keys} data={[]} />);

    keys.forEach((key) => {
      expect(getByText(key)).toBeInTheDocument();
    });
  });

  it("should render a list of rows of the given data", () => {
    const { getByText } = render(<Table keys={keys} data={data} />);

    data.forEach((breed) => {
      Object.keys(breed).forEach((key) => {
        expect(getByText(String(breed[key]))).toBeInTheDocument();
      });
    });
  });

  it("should sort the list of items by the specified key", () => {
    const { getByText, queryAllByTestId } = render(
      <Table keys={keys} data={data} />
    );

    const idHeader = getByText("id");

    fireEvent.click(idHeader);

    const rows = queryAllByTestId("row");
    expect(rows[0].innerHTML.includes("123")).toEqual(true);
  });

  it("should sort the list of items by the in a descending direction", () => {
    const { getByText, queryAllByTestId } = render(
      <Table keys={keys} data={data} />
    );

    const idHeader = getByText("id");

    fireEvent.click(idHeader);
    fireEvent.click(idHeader);

    const rows = queryAllByTestId("row");
    expect(rows[0].innerHTML.includes("231")).toEqual(true);
  });
});
