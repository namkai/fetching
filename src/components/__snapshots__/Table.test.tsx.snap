// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm sc-AxhCb jhFSQy"
  >
    <div
      class="sc-AxjAm sc-AxirZ iGwWVY"
    >
      <div
        class="sc-AxjAm sc-AxiKw sc-AxhUy gmDDPU"
      >
        id
      </div>
      <div
        class="sc-AxjAm sc-AxiKw sc-AxhUy gmDDPU"
      >
        name
      </div>
      <div
        class="sc-AxjAm sc-AxiKw sc-AxhUy gmDDPU"
      >
        life_span
      </div>
    </div>
  </div>
</DocumentFragment>
`;
